bootstrap/pkg/kfapp/coordinator/coordinator.go

```diff
@@ -528,14 +528,18 @@ func backfillKfDefFromInitOptions(kfdef *kfdefsv2.KfDef, options map[string]inte
 
 	// For boolean options there is no way to test whether they have been explicitly set in KfDef or
 	// not so we always override the value with the command line flag.
-	if options[string(kftypes.USE_BASIC_AUTH)] != nil {
-		kfdef.Spec.UseBasicAuth = options[string(kftypes.USE_BASIC_AUTH)].(bool)
-	}
-	if options[string(kftypes.SKIP_INIT_GCP_PROJECT)] != nil {
-		kfdef.Spec.SkipInitProject = options[string(kftypes.SKIP_INIT_GCP_PROJECT)].(bool)
-	}
-	if options[string(kftypes.DELETE_STORAGE)] != nil && kfdef.Spec.Platform == kftypes.GCP {
-		kfdef.Spec.DeleteStorage = options[string(kftypes.DELETE_STORAGE)].(bool)
+	// TODO(lunkai): I think we shouldn't backfill bool flags when using --config
```

Comment:
I think you are right. If CONFIG is specified everything should be in the config and we no longer need to backfill the flags.

I noted in the issue that we should probably raise an error if additional flags are specified with --config.
We could do that in a follow on PR.